risc0/zkvm/src/host/api/tests.rs

```diff
@@ -110,15 +114,25 @@ fn with_server<T, F: FnOnce() -> Result<T>>(addr: SocketAddr, f: F) -> Result<T>
 }
 
 #[test]
-fn execute_basic() {
+fn execute_elf() {
     let input = to_vec(&MultiTestSpec::DoNothing).unwrap();
     let env = ExecutorEnv::builder().add_input(&input).build().unwrap();
     let binary = Binary::new_elf_path(MULTI_TEST_PATH);
     TestClient::new().execute(env, binary).unwrap();
 }
 
 #[test]
-fn prove_basic() {
+fn execute_image() {
+    let input = to_vec(&MultiTestSpec::DoNothing).unwrap();
+    let env = ExecutorEnv::builder().add_input(&input).build().unwrap();
+    let program = Program::load_elf(&MULTI_TEST_ELF, GUEST_MAX_MEM as u32).unwrap();
+    let image = MemoryImage::new(&program, PAGE_SIZE as u32).unwrap();
+    let binary = image.into();
+    TestClient::new().execute(env, binary).unwrap();
```

Comment:
In CI, do we want to make sure these tests all run both with in-process execution and with sub-process execution? (Same question for prover tests).

Reply:
There are existing tests for the in-process which is actually more extensive. What I was shooting for in these tests is to get coverage for the client/server message passing specifically. Admittedly these tests are incomplete; there should be more to cover more cases.

I've been pondering how we could get a common collection of tests to be run in in-process or sub-process mode, but the best I can come up with is to use an integration test to ensure that `r0vm` gets built. These tests are a compromise, it doesn't actually spin up a separate process, it uses a separate thread instead. But it does use all the same messages.